ahmedeid6842/Math-Magicians
# Milestone: test components :ok_hand:

# Add tests to the Math Magicians app 🧮 :test_tube: 

🧪 We wrote unit tests for the Math Magicians app using the Jest and React Testing Library libraries. Specifically, we wrote unit tests for the files operate.js and calculate.js using Jest. We also created unit tests for all React components, using Jest snapshots to test the components. 🚀

## Done :heavy_check_mark: :memo: 
- [x] Write tests for the Calculator component to ensure correct functionality 🧮🧪
- [x] Verify that the Home page renders correctly and contains expected content 🏠🧪
- [x] Test the Quote page to ensure it renders correctly and displays a random quote 💬🧪
- [x] Perform unit tests on the calculate function to ensure it correctly calculates simple arithmetic expressions 🧮🧪
- [x] Verify the operate function performs the specified operation on two operands and handles errors gracefully 🧮🧪
- [x] Confirm that all tests pass successfully on local machine before submitting pull request ✔️
- [x] Add description of all tests to pull request 📝

## Test cases :man_technologist: 

1. `Calculator` component tests:
   - Verify that the component renders correctly using React Testing Library
   - Test that the display updates when buttons are clicked
   - Verify that basic arithmetic operations work correctly

2. `Home` page tests:
   - Test that the page renders correctly using Jest and React Testing Library
   - Verify that the page contains the expected content

3. `Quote` page tests:
   - Test that the page renders correctly using React Testing Library
   - Verify that the page displays a random quote

4. `calculate` function tests:
   - Test that the function correctly calculates the result of simple arithmetic expressions using Jest

5. `operate` function tests:
   - Test that the function performs the specified operation on two operands using Jest
   - Verify that the function handles errors gracefully

### All tests passed successfully on my local machine and have been included in this pull request. 🚀

![4zFo](https://github.com/ahmedeid6842/Math-Magicians/assets/57197702/5fdaa28e-207b-463b-bb22-17b19ceecf04)
